File: problem.set.1/problem.2.'CCC '02 J2 - AmeriCanadian'.py
consonants = ['b', 'c', 'd', 'f', 'g', 'j', 'k' , 'l', 'm' , 'n', 'p', 'q', 's', 't', 'v', 'x', 'z', 'h', 'r', 'w']

while 1:
    word = input()

    if word == 'quit!':
        break
  
    if len(word) > 4: # word needs to be longer than 4 characters long accordinqg to problem statement
        if word[-3] in consonants and word[-2] == 'o' and word[-1] == 'r': # find -(constant)or suffix
            word = word[:-2] + 'our' # replace with -our suffix
    
    print(word)

File: problem.set.1/problem.2.'CCC '02 J2 - AmeriCanadian'.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Array of consonants
        char[] consonants = {'b', 'c', 'd', 'f', 'g', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 's', 't', 'v', 'x', 'z', 'h', 'r', 'w'};

        while (true) {
            String word = scanner.nextLine();

            // Break if the word is 'quit!'
            if (word.equals("quit!")) {
                break;
            }

            // Check if the word has more than 4 characters
            if (word.length() > 4) {
                // Check if the word ends with a consonant followed by 'or'
                char thirdLastChar = word.charAt(word.length() - 3);
                if (isConsonant(thirdLastChar, consonants) && word.endsWith("or")) {
                    // Replace the suffix with 'our'
                    word = word.substring(0, word.length() - 2) + "our";
                }
            }

            // Output the modified word
            System.out.println(word);
        }

        scanner.close();
    }

    // Helper function to check if a character is a consonant
    public static boolean isConsonant(char ch, char[] consonants) {
        for (char consonant : consonants) {
            if (ch == consonant) {
                return true;
            }
        }
        return false;
    }
}